src/threadwrappers.cpp

```diff
@@ -111,7 +111,7 @@ extern "C" int __clone(int (*fn) (void *arg), void *child_stack, int flags,
   if (tid == -1) {
     JTRACE("Clone call failed")(JASSERT_ERRNO);
     ThreadSync::decrementUninitializedThreadCount();
-    delete thread;
+    JALLOC_HELPER_FREE(thread);
```

Comment:
Good catch!  @rohgarg, the issue is that the function for this code had previously called `getNewThread()`, which got a thread from the `threads_freelist`.  We should give it back to the free list.  By calling `threadIsDead()`, we will put it back on the free list.